src/PhpDoc/TypeNodeResolver.php

```diff
@@ -360,7 +360,7 @@ private function resolveIdentifierTypeNode(IdentifierTypeNode $typeNode, NameSco
 				return new CallableType(null, null, true, null, null, [], TrinaryLogic::createYes());
 
 			case 'pure-closure':
-				return new ClosureType();
+				return new ClosureType(null, null, true, null, null, null, [], [], []);
```

Comment:
I just realized how we could make this nicer - create `public static function createPure(): self`.